control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -218,11 +218,28 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
                     })),
                 );
 
+                let volume_group = context.volume_group();
+
                 IoEngineApiClient::get_client()
-                    .create_volume(&u, replica_count, size, volume_topology, thin)
+                    .create_volume(
+                        &u,
+                        replica_count,
+                        size,
+                        volume_topology,
+                        thin,
+                        volume_group.clone().map(VolumeGroup::new),
+                    )
                     .await?;
 
-                debug!(volume.uuid = volume_uuid, "Volume successfully created");
+                if volume_group.is_some() {
```

Comment:
if let Some(.... and avoid the clone&unwrap below..
Also, could move this trace to the actual `create_volume` call?
Another point, don't we already have a debug volume created trace just below? Please make sure we don't keep duplicates.

Reply:
Done.